plugins/hfp_ag_bluez5.c

```diff
@@ -111,8 +128,35 @@ static DBusMessage *profile_new_connection(DBusConnection *conn,
 	}
 
 	modem = modems->data;
+
 	DBG("Picked modem %p for emulator", modem);
 
+	memset(&saddr, 0, sizeof(saddr));
+	optlen = sizeof(saddr);
+
+	if (getsockname(fd, (struct sockaddr *) &saddr, &optlen) < 0) {
+		err = errno;
+		ofono_error("RFCOMM getsockname(): %s (%d)", strerror(err),
+									err);
+		close(fd);
```

Comment:
Why we close the fd when we return with error but we don't when all was good even when we duplicate the fd in that case? Does dbus_message_new_method_return() close the fd but dbus_message_new_method_return() not?